Tested this PR.
Repetier v1.0.6 translates `M110 N100` to `N100 M110`. So no danger from that side.
Pronterface 2014.03.10:

```
>>>M110 N100
SENDING:M110 N100
Error:No Checksum with line number, Last Line: 0
[ERROR] Error:No Checksum with line number, Last Line: 0
Resend: 1
```

looks not very promising.
Octoprint translates `M110 N100` to: 

```
Send: N100 M110 N100*125
Recv: ok
```

--- a/Marlin/Marlin_main.cpp
+++ b/Marlin/Marlin_main.cpp
@@ -817,7 +817,7 @@ void get_command() {
           }
           // if no errors, continue parsing
         }
-        else {
+        else if (npos == command) {
           gcode_line_error(PSTR(MSG_ERR_NO_CHECKSUM));
           return;
         }

